src/System.Management.Automation/engine/parser/FusionAssemblyIdentity.cs

```diff
@@ -210,10 +204,7 @@ internal static unsafe Version GetVersion(IAssemblyName nameObject)
             uint result;
             uint size = sizeof(uint);
             int hr = nameObject.GetProperty(propertyId, &result, ref size);
-            if (hr != 0)
-            {
-                throw Marshal.GetExceptionForHR(hr);
-            }
+            Marshal.ThrowExceptionForHR(hr);
```

Comment:
It seems we need to have `if (hr != 0)` because the method checks `error < 0` https://source.dot.net/#System.Private.CoreLib/Marshal.cs,604